File: Prog/Find_winner_of_circular_game/Find_Winner.py
def find_winner(n, k):
    winner = 0  # The winner's index
    for i in range(1, n + 1):
        winner = (winner + k) % i
    return winner + 1  # Adding 1 to convert index to friend number


n1 = int(input("Enter the value of n: "))
k1 = int(input("Enter the value of k: "))

winner = find_winner(n1, k1)
print("Winner for n =", n1, ", k =", k1, "is:", winner)

File: Prog/Find_winner_of_circular_game/Find_Winner.cpp
#include <iostream>

int findWinner(int n, int k) {
    int winner = 0;  // The winner's index
    for (int i = 1; i <= n; i++) {
        winner = (winner + k) % i;
    }
    return winner + 1;  // Adding 1 to convert index to friend number
}

int main() {
    int n, k;
    std::cout << "Enter the value of n: ";
    std::cin >> n;
    std::cout << "Enter the value of k: ";
    std::cin >> k;

    int winner = findWinner(n, k);
    std::cout << "Winner for n = " << n << ", k = " << k << " is: " << winner << std::endl;

    return 0;
}

